src/cycax/cycad/engines/simple_2d.py

```diff
@@ -140,19 +140,19 @@ def _figure_feature(self, ax: matplotlib.axes._axes.Axes, feature: dict):
         elif feature_type == "hole":
             self._hole(ax, feature)
 
-    def save_as_figure(self):
+    def build(self):
         """This method will coordinate the drawing and saving of the figure that is being decoded from the provided json."""
-        in_name = os.getcwd() + "/" + self.part_no + "/" + self.part_no + ".json"
+        in_name = "{cwd}/{name}/{name}.json".format(cwd=os.getcwd(), name=self.name)
         with open(in_name) as f:
             data = json.load(f)
         fig, ax = plt.subplots()
-        for feature in data:
+        for feature in data["parts"]:
             if feature["type"] == "add":
-                self._bounding_box(feature)
-            self._figure_feature(ax, feature)
-        ax.set_title(self.part_no)
+                self.bounding_box(feature)
+            self.figure_feature(ax, feature)
+        ax.set_title(self.name)
         ax.autoscale_view()
         ax.set_aspect("equal", "box")
-        figfile = os.getcwd() + "/" + self.part_no + "/" + self.part_no + "-fig.svg"
+        figfile = os.getcwd() + "/" + self.name + "/" + self.name + "-fig.svg"
```

Comment:
Lets use the configured base_path rather than os.getcwd.
e.g. `figfile =  self._base_path / name / (name + "-fig.svg")`

I'm not sure what "-fig" in the name means. Maybe "-s2d" is more specific.